solutions/swb-app/src/dataSetRoutes.ts

```diff
@@ -148,12 +148,12 @@ export function setUpDSRoutes(router: Router, dataSetService: DataSetPlugin): vo
     '/projects/:projectId/datasets',
     wrapAsync(async (req: Request, res: Response) => {
       const authenticatedUser = validateAndParse<AuthenticatedUser>(AuthenticatedUserParser, res.locals.user);
-      const pageSize = toNumber(req.query.pageSize) || DEFAULT_API_PAGE_SIZE;
```

Comment:
Nice catch! For those wondering why this didnt work, if `toNumber(req.query.pageSize)` resolves to `0` then it will take the right side of the `||` statement because `0` is falsy in JS. If youre trying to check if the left side evaluates to `undefined` or `null` then `??` is the operator youre looking for, otherwise a ternary like used below is the way to go.